### PR TITLE
Try synchronizing dispose and output

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Try resolving https://github.com/dotnet/aspnetcore/issues/20336 with more synchronization.

### CODE REVIEW

File: src/Grpc/test/InteropTests/InteropTests.cs [hunk 1]
```diff
@@ -27,6 +27,7 @@ public InteropTests(ITestOutputHelper output)
         }
 
         [Theory]
+        [Repeat]
```
Comment (JunTaoLuo) [inline src/Grpc/test/InteropTests/InteropTests.cs]: Runs 10 times which should take an extra 20 mins.

**Comment by davidfowl** [comment]
Didn't we have a PR that fixed this by booting up the server per test.?

**Comment by JunTaoLuo** [comment]
Yea last time I saw
```
Attachments:
  /private/tmp/helix/working/AEED0942/w/A8390952/e/TestResults/5d20590f-dca0-40a5-9110-acab59d28ae2/Sequence_47afb6e8ca4d447884f4db3f416ebf35.xml
Test Run Aborted.
Total tests: Unknown
     Passed: 13
 Total time: 1.6884 Minutes
```
So I thought the server was logging in between tests, which I still think occured, which is why I converted to booting and shutting down a server per test. However, today @Tratcher pointed to a build that still failed and after looking at it, it seems like the server was logging after shutting down which can also cause this error. I've also added `Repeat` so if this is still flaky, we'll know sooner.

TL:DR my fix was insufficient. Hopefully this makes it more reliable.

**Comment by JamesNK** [comment]
What about explicitly unregistering test output logging from the processes in dispose?

**Comment by Tratcher** [comment]
> What about explicitly unregistering test output logging from the processes in dispose?

ProcessEx already does that, but still seems to be hitting a race.

File: src/Grpc/test/InteropTests/InteropTests.cs (hunk 1 above)
Comment (Tratcher) [reply src/Grpc/test/InteropTests/InteropTests.cs]: You'll run the PR build a few times with this and then remove it before checking in, right?

**Review by Tratcher — APPROVED** [review]

File: src/Grpc/test/InteropTests/InteropTests.cs (hunk 1 above)
Comment (JunTaoLuo) [reply src/Grpc/test/InteropTests/InteropTests.cs]: That's the plan.

**Comment by JunTaoLuo** [comment]
40 runs no failure, I think that's sufficient to declare this battle won.

**Comment by JunTaoLuo** [comment]
@anurse should test flakiness improvements like this go into preview3?
